### ISSUE TITLE
Distinguishing between different types of links on Windows

### ISSUE THREAD

**Issue by oliversalzburg** [nodejs/help#1520]
* **Node.js Version**: 8.12.0
* **OS**: Windows 10
* **Scope (install, code, runtime, meta, other?)**: runtime
* **Module (and version) (if relevant)**: fs

On Windows, when we create a symlink, we are given the options to create a file or directory symlink or a junction. How can I determine if an *existing* entity is a file or directory symlink or a junction?

[`stats.isSymbolicLink()`](https://nodejs.org/api/fs.html#fs_stats_issymboliclink) will return `true` for all 3 types and I did not find an API that would provide more detail.

This is in reference to: https://github.com/jprichardson/node-fs-extra/issues/626

**Comment by benjamingr** [comment]
@nodejs/windows

**Comment by Hakerh400** [comment]
> @nodejs/windows

According to the [collaborator guide](https://github.com/nodejs/node/blob/master/COLLABORATOR_GUIDE.md#who-to-cc-in-the-issue-tracker), the team name is `@nodejs/platform-windows`

**Comment by benjamingr** [comment]
@Hakerh400 you're right, I'm just on holiday and not on my PC :) @nodejs/platform-windows for an actual ping. Thanks for catching this!

**Comment by tniessen** [comment]
I'm not sure whether it makes sense to implement this. Anyway, all three types are probably of type `FILE_ATTRIBUTE_REPARSE_POINT` and [this document](https://docs.microsoft.com/de-de/windows/desktop/FileIO/determining-whether-a-directory-is-a-volume-mount-point) might be relevant for distinguishing them.

**Comment by oliversalzburg** [comment]
I have already implemented this at https://github.com/oliversalzburg/is-junction

**Comment by tniessen** [comment]
@oliversalzburg You seem to be doing exactly what I suggested :) Is this issue resolved then?

**Comment by oliversalzburg** [comment]
@tniessen I was hoping for a solution in Node itself 😢

**Comment by gireeshpunathil** [comment]
@oliversalzburg - is this resolved? please feel to contribute as a PR, if you have specific proposals to improve!

**Comment by oliversalzburg** [comment]
@gireeshpunathil I have no reason to believe that it was resolved and nobody qualified ever responded to my inquires.

**Comment by gireeshpunathil** [comment]
@oliversalzburg - I checked the code and see what you are saying. Basically it is a good to have feature (to further specialize on the link info); but prospects of being implemented in core may depend on strength of the use case and number of people asking for it, and hence I suggested a PR in the previous comment.

**Comment by tniessen** [comment]
> nobody qualified ever responded to my inquires.

@oliversalzburg This kind of makes it sound as if the person who did respond, me, is unqualified. I will kindly assume that you did not actually mean to imply that.

I opened https://github.com/libuv/libuv/issues/2384. libuv is responsible for platform-specific IO.

**Comment by oliversalzburg** [comment]
@tniessen  I posted here in the hope to learn how to achieve this in NodeJS, because I didn't think that this is impossible. I didn't read your response as a definitive "This is currently not possible", so I assumed we were still waiting for someone who knows how to do it.

Given that months had passed since I asked about this, I already went ahead and wrote the module I linked, which wasn't an acceptable solution for the node-fs-extra members and the issue remains unresolved.

I doubt many people would ask for the ability to distinguish link types explicitly, but it ultimately leads to extremely hard to debug issues when using the affected code in node-fs-extra.

Thanks for opening the linked issue.
